From 37b01b00d37e1ce7e2f1de83ef1d96c65c1bd42d Mon Sep 17 00:00:00 2001
From: OneRecovery Build <build@onerecovery.org>
Date: Tue, 19 Mar 2025 00:00:00 +0000
Subject: [PATCH] Fix ABI header issues

Ensure tools ABI headers match their arch counterparts to prevent
ABI header mismatch warnings during compilation.

This patch creates symlinks for the problematic header files:
- arch/x86/lib/insn.c
- arch/x86/include/asm/inat.h
- arch/x86/include/asm/insn.h 
- arch/x86/lib/inat.c

The symlinks replace the copies in the tools/ directory to ensure
they stay in sync.
---
 tools/arch/x86/include/asm/inat.h | 0
 tools/arch/x86/include/asm/insn.h | 0
 tools/arch/x86/lib/inat.c         | 0
 tools/arch/x86/lib/insn.c         | 0
 4 files changed, 0 insertions(+), 0 deletions(-)
 create mode 120000 tools/arch/x86/include/asm/inat.h
 create mode 120000 tools/arch/x86/include/asm/insn.h
 create mode 120000 tools/arch/x86/lib/inat.c
 create mode 120000 tools/arch/x86/lib/insn.c

diff --git a/tools/arch/x86/include/asm/inat.h b/tools/arch/x86/include/asm/inat.h
new file mode 120000
index 000000000000..62af9aff9acf
--- /dev/null
+++ b/tools/arch/x86/include/asm/inat.h
@@ -0,0 +1 @@
+../../../../arch/x86/include/asm/inat.h
\ No newline at end of file
diff --git a/tools/arch/x86/include/asm/insn.h b/tools/arch/x86/include/asm/insn.h
new file mode 120000
index 000000000000..6cf1a5022747
--- /dev/null
+++ b/tools/arch/x86/include/asm/insn.h
@@ -0,0 +1 @@
+../../../../arch/x86/include/asm/insn.h
\ No newline at end of file
diff --git a/tools/arch/x86/lib/inat.c b/tools/arch/x86/lib/inat.c
new file mode 120000
index 000000000000..0edc71258533
--- /dev/null
+++ b/tools/arch/x86/lib/inat.c
@@ -0,0 +1 @@
+../../../../arch/x86/lib/inat.c
\ No newline at end of file
diff --git a/tools/arch/x86/lib/insn.c b/tools/arch/x86/lib/insn.c
new file mode 120000
index 000000000000..b3a42ebbf4f2
--- /dev/null
+++ b/tools/arch/x86/lib/insn.c
@@ -0,0 +1 @@
+../../../../arch/x86/lib/insn.c
\ No newline at end of file
-- 
2.39.3